src/chars.rs

```diff
@@ -128,15 +128,29 @@ pub(crate) fn count_impl<T: ByteChunk>(text: &[u8]) -> usize {
             inv_count += ((byte & 0xC0) == 0x80) as usize;
         }
 
-        // Take care of the middle bytes in big chunks.
-        for chunks in middle.chunks(T::MAX_ACC) {
-            let mut acc = T::zero();
-            for chunk in chunks.iter() {
-                acc = acc.add(chunk.bitand(T::splat(0xc0)).cmp_eq_byte(0x80));
+        let char_boundaries = |val: &T| val.bitand(T::splat(0xc0)).cmp_eq_byte(0x80);
+
+        // Take care of the middle bytes in big chunks. Loop unrolled
+        for chunks in middle.chunks_exact(4) {
+            let mut iter = chunks.iter();
+            while let Some(chunk) = iter.next() {
+                let val1 = char_boundaries(chunk);
+                let val2 = char_boundaries(iter.next().unwrap());
+                let val3 = char_boundaries(iter.next().unwrap());
+                let val4 = char_boundaries(iter.next().unwrap());
+                let val1_2 = val1.add(val2);
+                let val3_4 = val3.add(val4);
+                inv_count += val1_2.add(val3_4).sum_bytes();
             }
-            inv_count += acc.sum_bytes();
         }
 
+        // Take care of the rest of the chunk
+        let mut acc = T::zero();
+        for chunk in middle.chunks_exact(4).remainder() {
+            acc = acc.add(char_boundaries(chunk));
+        }
+        inv_count += acc.sum_bytes();
+
```

Comment:
This is the only change that I'm feeling a little stuck on.  I follow what's going on here, but I don't think it's clear why this is faster on ARM:

```
for chunk in middle {
    inv_count += chunk.bitand(T::splat(0xc0)).cmp_eq_byte(0x80).sum_bytes();
}
```

...than the existing code.  Because this new version requires doing `sum_bytes()` more often, which (in theory) should make it slower.

I'm not doubting that the new version is faster on ARM, but I think figuring out *why* that's the case deserves some investigation before committing to the code change.  Especially since it results in an expected regression on x86 (presumably due to the additional `sum_bytes()` calls) which needs to be mitigated with explicit loop unrolling.